File: backend/src/main/java/com/ead/backend/entity/ServiceCenter.java
package com.ead.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.UUID;

@Entity
@Table(name = "service_center")
@Data
@NoArgsConstructor
public class ServiceCenter {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(columnDefinition = "UUID")
    private UUID id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String address;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false, precision = 10, scale = 8)
    private BigDecimal latitude;

    @Column(nullable = false, precision = 11, scale = 8)
    private BigDecimal longitude;

    private String phone;
    private String email;

    @Column(name = "operating_hours", columnDefinition = "TEXT")
    private String operatingHours;

    @Column(nullable = false)
    private Boolean isActive = true;

    @Column(name = "center_slot", nullable = false)
    private Integer centerSlot = 2; // Default to 2 slots
}

File: frontend/src/types/serviceCenter.ts
import type { LatLngExpression } from "leaflet";

export interface ServiceCenter {
  id: string; // UUID from backend
  name: string;
  address: string;
  city: string;
  latitude: number;
  longitude: number;
  phone: string;
  email: string;
  operatingHours: string;
  isActive: boolean;
}

export interface UserLocation {
  latitude: number;
  longitude: number;
  accuracy: number;
}

export type MapCenter = LatLngExpression;